src/Spinner/Spinner.tsx

```diff
@@ -1,45 +1,57 @@
 import React from 'react'
 import styled from 'styled-components'
 import sx, {SxProp} from '../sx'
-import {ComponentProps} from '../utils/types'
+import {VisuallyHidden} from '../internal/components/VisuallyHidden'
+import {HTMLDataAttributes} from '../internal/internal-types'
+import {Box} from '../'
 
 const sizeMap = {
   small: '16px',
   medium: '32px',
   large: '64px',
 }
 
-export interface SpinnerInternalProps {
+export type SpinnerProps = {
   /** Sets the width and height of the spinner. */
   size?: keyof typeof sizeMap
-}
+  /** Sets the text conveyed by assistive technologies such as screen readers. Set to `null` if the loading state is displayed in a text node somewhere else on the page. */
+  srText?: string | null
+  /** @deprecated Use `srText` instead. */
+  'aria-label'?: string | null
+} & HTMLDataAttributes &
+  SxProp
 
-function Spinner({size: sizeKey = 'medium', ...props}: SpinnerInternalProps) {
+function Spinner({size: sizeKey = 'medium', srText = 'Loading', 'aria-label': ariaLabel, ...props}: SpinnerProps) {
   const size = sizeMap[sizeKey]
+  const hasSrAnnouncement = Boolean(srText || ariaLabel)
 
   return (
-    <svg height={size} width={size} viewBox="0 0 16 16" fill="none" {...props}>
-      <circle
-        cx="8"
-        cy="8"
-        r="7"
-        stroke="currentColor"
-        strokeOpacity="0.25"
-        strokeWidth="2"
-        vectorEffect="non-scaling-stroke"
-      />
-      <path
-        d="M15 8a7.002 7.002 0 00-7-7"
-        stroke="currentColor"
-        strokeWidth="2"
-        strokeLinecap="round"
-        vectorEffect="non-scaling-stroke"
-      />
-    </svg>
+    /* inline-flex removes the extra line height */
+    <Box sx={{display: 'inline-flex'}} role={hasSrAnnouncement ? 'status' : undefined}>
```

Comment:
Would we want to add the `role` value as a prop instead of a static value? If so, we would only have two values to pick from, `status` and `alert`, with `status` being the default value.